src/WebSockets.jl

```diff
@@ -381,27 +381,24 @@ function handle_control_frame(ws::WebSocket, wsf::WebSocketFragment)
         @debug ws, " received OPCODE_PING"
         # Nothing to do here; no reply is needed for a pong message.
     else  # %xB-F are reserved for further control frames
-        error(" while handle_control_frame(ws|$(ws.server ? "server" : "client"), wsf): Unknown opcode $(wsf.opcode)")
+        error("while handle_control_frame(ws|$(ws.server ? "server" : "client"), wsf): Unknown opcode $(wsf.opcode)")
     end
 end
 
 """ Read a frame: turn bytes from the websocket into a WebSocketFragment."""
 function read_frame(ws::WebSocket)
-    # Try to read two bytes. There is no guarantee that two bytes are actually allocated.
-    ab = read(ws.socket, 2)
-    #=
-    Browsers will seldom close in the middle of writing to a socket,
-    but other clients often do, and the stacktraces can be very long.
-    ab can be assigned, but of length 1. Use an enclosing try..catch in the calling function
-    =#
-    a = ab[1]
+    if eof(ws.socket)
+        throw(WebSocketError(1006, "Client side closed socket connection"))
+    end
+    # Read first byte
+    a = read(ws.socket, UInt8)
     fin    = (a & 0b1000_0000) >>> 7  # If fin, then is final fragment
     rsv1   = a & 0b0100_0000  # If not 0, fail.
     rsv2   = a & 0b0010_0000  # If not 0, fail.
     rsv3   = a & 0b0001_0000  # If not 0, fail.
     opcode = a & 0b0000_1111  # If not known code, fail.
 
-    b = ab[2]
+    b = read(ws.socket, UInt8)
```

Comment:
This line is quite finicky, I remember some quite unexpected behaviour when trying to change it earlier. It's actually part of the motivation for having the very large test suite now. We want to trigger those weird edge cases by testing very thoroughly.

Now that we have extensive tests, and they don't fail, we could make this change I suppose. Unfortunately the benchmark tests were never really completed so would the two read operations be faster? Is there any other motivation than beautification?